app/controllers/cm_admin/resource_controller.rb

```diff
@@ -0,0 +1,199 @@
+        else
+          format.html { render '/cm_admin/main/new', notice: "#{action_name.titleize} #{@ar_object.class.name.downcase} is unsuccessful" }
+        end
+      end
+    end
+
+    def custom_controller_action(action_name, params)
+      current_action = CmAdmin::Models::Action.find_by(@model, name: action_name.to_s)
+      if current_action
+        @current_action = current_action
+        @ar_object = @model.ar_model.name.classify.constantize.find(params[:id])
+        if @current_action.child_records
+          child_records = @ar_object.send(@current_action.child_records)
+          @associated_model = CmAdmin::Model.find_by(name: @model.ar_model.reflect_on_association(@current_action.child_records).klass.name)
+          if child_records.is_a? ActiveRecord::Relation
+            @associated_ar_object = filter_by(params, child_records)
+          else
+            @associated_ar_object = child_records
+          end
+          return @ar_object, @associated_model, @associated_ar_object
+        end
+        return @ar_object
+      end
+    end
+
+    def filter_by(params, records, filter_params={}, sort_params={})
+      filtered_result = OpenStruct.new
+      sort_column = "created_at"
+      sort_direction = %w[asc desc].include?(sort_params[:sort_direction]) ? sort_params[:sort_direction] : "asc"
+      sort_params = {sort_column: sort_column, sort_direction: sort_direction}
+      
+      records = "CmAdmin::#{@model.name}Policy::Scope".constantize.new(Current.user, @model.name.constantize).resolve if records.nil?
+      records = records.order("#{@current_action.sort_column} #{@current_action.sort_direction}")
+
+      final_data = CmAdmin::Models::Filter.filtered_data(filter_params, records, @model.filters)
+      pagy, records = pagy(final_data)
+      filtered_result.data = records
+      filtered_result.pagy = pagy
+      # filtered_result.facets = paginate(page, raw_data.size)
+      # filtered_result.sort = sort_params
+      # filtered_result.facets.sort = sort_params
+      return filtered_result
+    end
+
+    def resource_params(params)
+      permittable_fields = @permitted_fields || @model.ar_model.columns.map(&:name).reject { |i| CmAdmin::REJECTABLE_FIELDS.include?(i) }.map(&:to_sym)
+      permittable_fields += @model.ar_model.name.constantize.reflect_on_all_associations.map {|x|
+        next if x.options[:polymorphic]
+        # The first if statement is added for compatibilty with cm-page-builder.
+        # One of the associated models of cm-page-builder was throwing error.
+        # The associated model is CmPageBuilder::Rails::PageComponent.
+        # When using reflection, the value of klass of the above association is uninitialized.
+        # As a result, it was throwing error in the 2nd elsif statement.
+        if x.name == :page_components
+          x.name
+        elsif x.class.name.include?('HasOne')
```

Comment:
@anbublacky - We can remove the hack for page builder now.
```suggestion
        if x.class.name.include?('HasOne')
```